apps/artemis-web/pages/api/userlist.ts

```diff
@@ -0,0 +1,28 @@
+import nc from 'next-connect';
+import {
+  NextApiRequestExtended,
+  NextApiResponseExtended,
+} from '../../definitions';
+import auth from '../../middleware/auth';
+
+const handler = nc()
+  .use(auth)
+  .get(async (req: NextApiRequestExtended, res: NextApiResponseExtended) => {
+    if (!req.user || req.user.role !== 'admin') {
```

Comment:
hmm, is it possible to add this check in the .use chain like a middleware?
something like:
```
.use(auth)
.use(authorization(['admin']))
.get(...)
```

Reply:
Nice idea! Simple and elegant.